packages/driver/src/cy/commands/querying.js

```diff
@@ -407,7 +409,11 @@ module.exports = (Commands, Cypress, cy) => {
         filter = ''
       }
 
-      _.defaults(options, { log: true })
+      if (options.matchCase === true && _.isRegExp(text) && text.flags.includes('i')) {
+        throw new Error('cy.contains() content has i flag and matchCase is true. What is intended?')
```

Comment:
Let's move this into `error_messages.coffee` and use `$utils.throwErrByPath`. The goal is to have all error messages in there. Also, let's change the text to "You passed a regular expression with the case-insensitive (i) flag and `{ matchCase: true }` to cy.contains(). Those options conflict with each other, so please choose one or the other.`